node/gum/proc-macro/src/lib.rs

```diff
@@ -43,6 +43,27 @@ pub fn warn(item: proc_macro::TokenStream) -> proc_macro::TokenStream {
 	gum(item, Level::Warn)
 }
 
+/// Print a warning or debug level message depending on their frequency
+#[proc_macro]
```

Comment:
can we rename this to something that gives an understanding on what frequent means ?
`warn_if_more_frequent_than_1s` since it might get used in other places as well.


Alternatively, you can make the threshold as an enum parameter as well, the call could be something like:
```
warn_if_frequent(
  frequency: freq,
  threshold:  SECONDS(1)
)
```

Reply:
I like the idea of being more explicit. Passing max rate to arguments seems more perspective. However, I'd like to ask @eskimor. Maybe it should be as short and predefined as possible.

Reply:
Explicitness is good. What is too frequent is very likely dependent on the concrete warning.